spinalcordtoolbox/utils/profiling.py

```diff
@@ -0,0 +1,51 @@
+import atexit
+import logging
+import time
+from argparse import Action, SUPPRESS
+
+PROFILING_TIMER = None
+
+
+# Wrapper class which manages our profiling timer
+class Timer:
+    def __init__(self):
+        self._t0 = time.time()
+        atexit.register(self.stop)
+
+    def __del__(self):
+        """
+        Prevent early deletion of Timers from breaking atexit
+        """
+        atexit.unregister(self.stop)
+
+    def stop(self):
+        time_delta = time.time() - self._t0
+        logging.warning(f"PROFILER: Elapsed time: {time_delta: .3f} seconds.")
```

Comment:
Tiny nit: Extra space!

![image](https://github.com/user-attachments/assets/d4abdbc1-78a5-417a-9d66-9e2b6141f60f)

```suggestion
    def stop(self):
        time_delta = time.time() - self._t0
        logging.warning(f"PROFILER: Elapsed time: {time_delta:.3f} seconds.")
```

I hadn't realized that adding a space within the `{}` expression would propagate outside of it? Kind of weird!

Reply:
As someone who possesses the Linux brainrot, I love the Unix time format. However, given the use-case for this profiling (namely, for comparing between different runs), having a format that is trivial to parse and compare programmatically seems more important than one that is pretty.

I will add the newlines and change the separator though, as those do fall in line with the intended use case.

Reply:
Sounds good! :)